### PR TITLE
CAS-300 Show Gender Identity and Sex in the Referrals report

### PR DESCRIPTION
**Pull request by muhammad-elabdulla**
This PR [CAS-300](https://dsdmoj.atlassian.net/browse/CAS-300) includes:

- Add new column Gender Identity
- Rename existing column Gender to Sex  

to the referrals report 

[CAS-300]: https://dsdmoj.atlassian.net/browse/CAS-300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/TransitionalAccommodationReferralReportGenerator.kt [hunk 1]
```diff
@@ -29,7 +29,8 @@ class TransitionalAccommodationReferralReportGenerator : ReportGenerator<
         },
         pncNumber = personInfo.tryGetDetails { it.pnc },
         crn = referralData.crn,
-        gender = personInfo.tryGetDetails { it.gender },
+        sex = personInfo.tryGetDetails { it.gender },
+        genderIdentity = personInfo.tryGetDetails { it.profile?.genderIdentity },
```
Comment (sudhakar-plr) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/TransitionalAccommodationReferralReportGenerator.kt:33]: The changes looks ok, but since the story(cas-300) refers 'Referal Summary' screen then i believe the `genderIdentify` needs to be populated as per existing logic below. Could you check Gareth about requirement?

```
genderIdentity = when (personInfoResult.offenderDetailSummary.offenderProfile.genderIdentity) {
        "Prefer to self-describe" -> personInfoResult.offenderDetailSummary.offenderProfile.selfDescribedGender
        else -> personInfoResult.offenderDetailSummary.offenderProfile.genderIdentity
      },

```
      
Refer PeopleController.peopleSearchGet() which is used to display `Referral Summary` page.

Comment (muhammad-elabdulla) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/TransitionalAccommodationReferralReportGenerator.kt]: I spoke with Gareth and we make it consistent with Referral Summary

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/OffenderDetailsUtils.kt [hunk 2]
```diff
@@ -101,7 +102,7 @@ fun CaseSummary.asOffenderDetailSummary() = OffenderDetailSummary(
     riskColour = null,
     disabilities = null,
     genderIdentity = this.profile?.genderIdentity,
-    selfDescribedGender = null,
+    selfDescribedGender = this.profile?.selfDescribedGender,
```
Comment (sudhakar-plr) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/OffenderDetailsUtils.kt:105]: is this change needed for our stories? if not, then i would recommend to leave it as it was before.

**Review by sudhakar-plr — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/OffenderDetailsUtils.kt (hunk 2 above)
Comment (muhammad-elabdulla) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/OffenderDetailsUtils.kt]: I set it as this value is coming from the profile like genderIdentity
